packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -780,33 +866,60 @@ class SliverAppBar extends StatelessWidget {
   /// remain visible rather than being scrolled out of view.
   final bool pinned;
 
+  /// If [snap] and [floating] are true then the floating app bar will "snap"
+  /// into view.
+  ///
+  /// If [snap] is true then a scroll that exposes the floating app bar will
+  /// trigger an animation that slides the entire app bar into view. Similarly if
+  /// a scroll dismisses the app bar, the animation will slide the app bar
+  /// completely out of view.
+  ///
+  /// Snapping only applies when the app bar is floating, not when the appbar
+  /// appears at the top of its scroll view.
+  final bool snap;
+
+  @override
+  _SliverAppBarState createState() => new _SliverAppBarState();
+}
+
+class _SliverAppBarState extends State<SliverAppBar> with TickerProviderStateMixin {
   @override
   Widget build(BuildContext context) {
-    final double topPadding = primary ? MediaQuery.of(context).padding.top : 0.0;
-    final double collapsedHeight = (pinned && floating && bottom != null)
-      ? bottom.bottomHeight + topPadding : null;
+    final double topPadding = config.primary ? MediaQuery.of(context).padding.top : 0.0;
+    final double collapsedHeight = (config.pinned && config.floating && config.bottom != null)
+      ? config.bottom.bottomHeight + topPadding : null;
+
+    FloatingHeaderSnapConfiguration snapConfiguration;
```

Comment:
since you're stateful anyway, you might as well cache this object, creating it in initState/didUpdateConfig. That way if you get built redundantly, you don't have to rebuild it.